searx/webapp.py

```diff
@@ -1135,9 +1130,37 @@ def __call__(self, environ, start_response):
         return self.app(environ, start_response)
 
 
+class InitializeApplication:
+    '''Check settings['server']['secret_key'] when the application is started
+    Allow to import searx.webapp without issue.
+    '''
+
+    def __init__(self, app):
+        self.app = app
+
+        # check secret_key
+        if not searx_debug and settings['server']['secret_key'] == 'ultrasecretkey':
+            logger.error('server.secret_key is not changed. Please use something else instead of ultrasecretkey.')
+            exit(1)
+
+        # initialize the engines except on the first run of the werkzeug server.
+        if not werkzeug_reloader\
+           or (werkzeug_reloader and os.environ.get("WERKZEUG_RUN_MAIN") == "true"):
+            initialize()
+
+    def __call__(self, environ, start_response):
+        return self.app(environ, start_response)
+
+
+# uwsgi application
 application = app
+
 # patch app to handle non root url-s behind proxy & wsgi
 app.wsgi_app = ReverseProxyPathFix(ProxyFix(application.wsgi_app))
 
+# initiliaze the webapp when it is started not loaded
+app.wsgi_app = InitializeApplication(app.wsgi_app)
```

Comment:
Just for my interest; is it typical design in  _middleware_ to chain? What might be the drawback (except that the class name is inappropriate) placing the implementation into the existing `ReverseProxyPathFix` class?

Reply:
It should be the right way
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.wsgi_app 

But anyway the line 1162 creates an instance of InitializeApplication, so the code in __init__ runs, so it doesn't provide anything. So I'm going to remove this class, and check at import time.

---

BTW starlette offers an `on_startup` option: https://www.starlette.io/applications/